src/main/python/image_preprocessing/face_manager.py

```diff
@@ -25,6 +25,30 @@ def __init__(self, shape_predictor_path):
         """
         self.shape_predictor_path = shape_predictor_path
 
+    def face_likeness_extraction(self, image):
+        """
+        This function find a face in the image passed and is optimised
+        to align the face before being returned.
+        Author(s):
+            Stephan Nell
+        Args:
+            image (:obj:'OpenCV image'): Image containing the face we need to detect and extract.
+        Raises:
+            ValueError: If no face can be detected.
+        Returns:
+            obj:'OpenCV image': Any background and unnecessary components are removed and only
+            the aligned face is returned in gray scale for facial likeness.
+        Todo:
+            Return error if no face detected
+
+        """
+        rectangle = self.detect(image)
+        predictor = dlib.shape_predictor(self.shape_predictor_path)
+        face_aligner = FaceAligner(predictor, desiredFaceWidth=256)
+        face_aligned = face_aligner.align(image, image, rectangle)
+        gray_face = cv2.cvtColor(face_aligned, cv2.COLOR_BGR2GRAY)
```

Comment:
I think we're misunderstanding each other. The existence of this function is fine, I am only questioning the very last line that converts image of the face to a grayscale image. In essence, as I understand it, this function does three (3) things:
1. Detect the face in the image.
2. Align the detected face.
3. Convert the face to grayscale and return it.

What I am asking is that you remove step 3 from this function, because a user of our API may think that this is a bug because it is an unwanted side-effect. Instead of converting the face to grayscale in this function here, rather convert it to grayscale in the lines that I mentioned below.

Correct me if I am wrong.
@Nicvaniek would you also just mind giving your opinion?